### PR TITLE
[6/signals] sigwinch-forward-unix: Forward SIGWINCH to child PTY

### LINKED ISSUE #59: [6/signals] sigwinch-forward-unix: Forward SIGWINCH to child PTY
**Task ID:** `sigwinch-forward-unix`
**Phase:** 6 &nbsp; **Area:** `signals` &nbsp; **Kind:** `impl`
**Tracked in:** #11

## Goal

self-pipe Resize → terminal::size() → master.resize() を実装。並行な resize の合流点と最後尾サイズの優先を保証。

## Definition of Done

- [ ] Implementation complete
- [ ] Unit / integration / E2E tests added or updated
- [ ] `cargo fmt --check` passes
- [ ] `cargo clippy --all-targets -- -D warnings` passes
- [ ] `cargo test --all-targets --all-features` passes
- [ ] Rubber-duck self-review pass complete (for non-trivial changes)

## Dependencies

**Blocked by:** #22 (`pty-spawn-child`)

**Blocks:** #61 (`e2e-sigwinch-forward`), #77 (`release-tag-v0-1-0`)

## Notes for AI agents

- Project uses GitHub Flow + Conventional Commits + signed commits.
- Allowlist for arming is fixed (not configurable): copilot, codex,
  claude, aichat, gemini, qwen, ollama.
- tracing env: `QORRECTION_LOG` (NOT `RUST_LOG`).
- See the meta tracking issue (#11) for full spec lock and
  approach.

## Claim protocol

When you start work on this issue:

1. Self-assign and add the `status/in-progress` label (remove
   `status/ready` or `status/blocked`).
2. Open a PR referencing this issue (`Closes #59`).
3. The PR moves the issue to `status/in-review`.
4. Always rebase onto `main`; never merge into the feature branch.

### PR DESCRIPTION
**Pull request by kurone-kito**
## Summary
- install a SIGWINCH-only self-pipe path for the Unix PTY session so resize delivery starts without changing SIGTERM semantics ahead of `#60`
- coalesce pending resize wake bytes on the existing supervisor tick and forward only the latest valid host terminal size to `master.resize()`
- add focused tests for the new install mode, resize coalescing, unusable runtime sizes, and resize-failure cleanup

Closes #59

## Follow-ups
- `#61` will add rexpect coverage for real WINCH delivery into the child
- `#60` remains the owner of graceful SIGTERM shutdown and raw-mode restoration semantics

## Summary by CodeRabbit

* **Refactor**
  * More efficient and reliable terminal resize handling: multiple queued resize signals collapse into a single host-size query and resize, and resizing is skipped when the runtime size is unavailable.
  * PTY size derivation improved to preserve existing child size when current runtime size is degenerate and to fall back only when needed.
* **Tests**
  * Added tests for batched resize signals, skipped resizes when size is unavailable, failure propagation during resize, and PTY size edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/kurone-kito/qorrection/pull/135?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by kurone-kito** [comment]
**IDD Live Status**

| Field | Value |
| --- | --- |
| Phase | F3 merged |
| Claim | codex-71d687f / 532bb7bc-5471-4ce7-a825-0979a52a37b6 |
| Branch | issue/59-6-signals-sigwinch-forward-unix |
| Last checked | 2026-05-17T03:10:55Z |
| Open blockers | none |
| Next action | F4 cleanup then F5 discover |
| Authoritative by | merge commit fb7dbef45c83d26114f9c6928fd41f087314857b on matched head fc20fe8f49be40de5f159b645191b44bd5d90608 |

**Comment by kurone-kito** [comment]
_codex-71d687f: review triage snapshot - IDD automation marker. Do not edit._

**Comment by kurone-kito** [comment]
_codex-71d687f: critique baseline - IDD automation marker. Do not edit._

File: src/pty/session.rs [hunk 1]
```diff
@@ -202,32 +235,79 @@ impl Drop for KillOnDropGuard {
...
-    let SpawnedSession {
-        child,
-        master: _master,
-    } = session;
-    let child = PtyChild { child };
-    run_pump_session_with(child, pump, Deadlines::production())
+    // after the supervisor returns.
+    #[cfg(unix)]
+    {
+        let signal_guard = SignalGuard::install_resize_only()?;
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/pty/session.rs]: Fixed in fc20fe8. I wrapped the Unix-only signal setup in `run_pump_session_unix_with_setup`, arm a `KillOnDropGuard` before calling `SignalGuard::install_resize_only()`, and only disarm that outer guard after setup succeeds and control is handed to the inner supervisor. I also added `signal_setup_failure_kills_child_before_supervisor_handoff` to lock the failure path down.

**Comment by kurone-kito** [comment]
advisory-wait: codex-71d687f fc20fe8f49be40de5f159b645191b44bd5d90608 2026-05-17T03:00:03Z

**Comment by kurone-kito** [comment]
_codex-71d687f: critique baseline - IDD automation marker. Do not edit._

**Comment by kurone-kito** [comment]
_codex-71d687f: review triage snapshot - IDD automation marker. Do not edit._

**Comment by kurone-kito** [comment]
**F4 Cleanup Evidence**

| Field | Value |
| --- | --- |
| Status | applied |
| Applied | 6 |
| Failed | 0 |
| Skipped | 0 |
| Permission-blocked | 0 |
| Notes | minimized CodeRabbit no-action summary as RESOLVED; minimized 5 stale operational markers as OUTDATED |
